model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java

```diff
@@ -949,6 +953,39 @@ public Stream<GroupModel> searchForGroupByNameStream(RealmModel realm, String se
         }).sorted(GroupModel.COMPARE_BY_NAME).distinct());
     }
 
+    @Override
+    public Stream<GroupModel> searchGroupsByAttributes(RealmModel realm, Map<String, String> attributes, Integer firstResult, Integer maxResults) {
+        Map<String, String> filteredAttributes = CollectionUtils.isEmpty(groupSearchableAttributes) ? attributes :
+                attributes.entrySet().stream().filter(m -> groupSearchableAttributes.contains(m.getKey()))
+                        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        CriteriaBuilder builder = em.getCriteriaBuilder();
+        CriteriaQuery<GroupEntity> queryBuilder = builder.createQuery(GroupEntity.class);
+        Root<GroupEntity> root = queryBuilder.from(GroupEntity.class);
+
+        List<Predicate> predicates = new ArrayList<>();
+
+        predicates.add(builder.equal(root.get("realm"), realm.getId()));
+
+        for (Map.Entry<String, String> entry : filteredAttributes.entrySet()) {
+            String key = entry.getKey();
```

Comment:
`key` may be `null` or empty, in which case the entry should be ignored.

Reply:
I can't seem to reply to the map implementation comment directly so replying here, we do have a map implementation and something is up with git history. I'll make sure it gets added back into the commits thanks for calling it out

Reply:
Added the logic to handle this case